### PR TITLE
Update dependencies.

### PR DESCRIPTION
**Pull request by MaximAlien**
Update Mapbox Maps to `v10.2.0-rc.1`, Navigation Native to `v81.0.0` and Common to `v21.0.0-rc.1`.

Also closing https://github.com/mapbox/mapbox-navigation-ios/issues/3479.

At this point there is a regression related to showing location indicator. It's being tracked here: https://github.com/mapbox/mapbox-maps-ios/issues/886.

Currently failing tests:
- ~~testRerouteAfterArrival~~
- testPausedFreeDrivePausesNavNativeNavigator
- testRouteSnappingOvershooting
- ~~testArrive~~
- ~~testPuck3DLayerPosition~~
- ~~testRunningAnimators~~
- ~~testTripPerWaypoint~~

### CODE REVIEW

**Comment by MaximAlien** [comment]
Update is not possible at this time due to:
```
error: Dependencies could not be resolved because root depends on 'mapbox-navigation-native-ios' 80.0.0..<81.0.0 and root depends on 'mapbox-maps-ios' 10.2.0-beta.1.
'mapbox-maps-ios' is incompatible with 'mapbox-navigation-native-ios' because 'mapbox-maps-ios' >= 10.2.0-beta.1 depends on 'mapbox-common-ios' 21.0.0-rc.1 and 'mapbox-navigation-native-ios' >= 80.0.0 depends on 'mapbox-common-ios' 20.1.0..<21.0.0.
```

Waiting for the latest version of Mapbox Navigation Native to proceed further.

File: Sources/MapboxCoreNavigation/BillingHandler.swift [hunk 1]
```diff
@@ -7,26 +7,21 @@ import os.log
 @_implementationOnly import MapboxCommon_Private
 
 /// BillingService from MapboxCommon
-private typealias BillingServiceNative = MapboxCommon_Private.BillingService
+private typealias BillingServiceNative = MapboxCommon_Private.BillingServiceFactory
```
Comment (MaximAlien) [inline Sources/MapboxCoreNavigation/BillingHandler.swift:10]: /cc @S2Ler, there are a couple of billing related changes.

Comment (S2Ler) [reply Sources/MapboxCoreNavigation/BillingHandler.swift:10]: Looks good.

File: Sources/MapboxCoreNavigation/BillingHandler.swift [hunk 2]
```diff
@@ -76,48 +71,48 @@ private final class ProductionBillingService: BillingService {
     }
 
     func getSKUTokenIfValid(for sessionType: BillingHandler.SessionType) -> String {
-        MapboxCommon_Private.BillingService.getSessionSKUTokenIfValid(for: tripSku(for: sessionType))
+        BillingServiceNative.getInstance().getSessionSKUTokenIfValid(for: tripSku(for: sessionType))
```
Comment (S2Ler) [inline Sources/MapboxCoreNavigation/BillingHandler.swift:74]: nit: I would extract `BillingServiceNative.getInstance()` into a computed property. 

```swift
var nativeBillingService: <WHAT TYPE HERE?> {
    BillingServiceNative.getInstance()
}
```

**Review by azarovalex — APPROVED** [review]
deps bump is ok

File: Cartfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-binary "https://api.mapbox.com/downloads/v2/carthage/mapbox-common/MapboxCommon.json" ~> 20.1.0
-binary "https://api.mapbox.com/downloads/v2/carthage/mobile-navigation-native/MapboxNavigationNative.xcframework.json" ~> 80.0
+binary "https://api.mapbox.com/downloads/v2/carthage/mapbox-common/MapboxCommon.json" == 21.0.0-rc.1
```
Comment (1ec5) [inline Cartfile:1]: Pinning this version is unfortunate but necessary for consistency [with SPM](https://github.com/mapbox/mapbox-navigation-native-ios/blob/beb14b30bdb3bcde32544756f1867d82b6d758bf/Package.swift#L8) and CocoaPods. But we’ll need to make sure to go back to a tadpole operator once MapboxCommon v21.0.0 is released and MapboxNavigationNative depends on it.

File: Sources/MapboxCoreNavigation/BillingHandler.swift (hunk 2 above)
Comment (1ec5) [reply Sources/MapboxCoreNavigation/BillingHandler.swift:74]: Nit: name it `shared`. 🙂

Comment (MaximAlien) [reply Sources/MapboxCoreNavigation/BillingHandler.swift:74]: Introduced `NativeBillingService.shared` for this purpose.

Comment (MaximAlien) [reply Sources/MapboxCoreNavigation/BillingHandler.swift]: I did some tests and see that `BillingServiceNative.getInstance()` somehow caches its instances (they're sometimes different, but still reused). Hopefully using only one shared instance won't cause any problems.

**Pull request merged**